### PR TITLE
Wrap the SDK in a main container `Archethic`

### LINKED ISSUE #27: Create class/container for any interaction
### Problem to solve

Everytime we need to interact with the blockchain, the functions requires to input the endpoint.
Which may be not really comfortable.

### Solution

So an idea would be to create a container or class to abstract the endpoint configuration and levereage this primary object to perform other queries, a bit like `web3`

You could have something like
```js
const archethic = new ARCHEthic("https://testnet.archethic.net")

const txBuilder = archethic.newTransactionBuilder("transfer")

const archethic.sendTransaction(tx)
```

### LINKED ISSUE #42: Add keychain transaction creation test
### Problem to solve

_No response_

### Solution

could you please add some tests to test these methods: 
- newAccessKeychainTransaction
- newKeychainTransaction
- getKeychain
these methods are here: https://github.com/archethic-foundation/libjs/blob/1e5e822788bf6c331c5b61a8c05da5801f5c9262/index.js#L176

### LINKED ISSUE #80: Add test for API calls
Add mock testing for api calls

### PR DESCRIPTION
**Pull request by samuelmanzanera**
The container is managing connection to the endpoint to retrieve the nearest nodes.

It also encapsulates submodules: Account, Network, Transaction Crypto and Utils are also exposed if needed

Fixes #27
Fixes #80
Fixes #42

### CODE REVIEW

**Review by Neylix — CHANGES_REQUESTED** [review]
I think in all projet we should use camelCase format,  there is some `token_id` `token_address` that remains

File: lib/api.js [hunk 1]
```diff
@@ -1,82 +1,147 @@
...
+import fetch from "cross-fetch";
+import { Socket } from "phoenix";
+import WebSocket from "isomorphic-ws";
+
+import { isHex, uint8ArrayToHex } from "./utils";
+
+//Use ES5 require, as ES6 import causes `Cannot instantiate an arrow function` error
+const withAbsintheSocket = require("@absinthe/socket");
+
+export function getNearestEndpoints(endpoint) {
```
[the remark this answers is not in the document]
Comment (samuelmanzanera) [reply lib/api.js:10]: Actually `async` is used when you use `await` inside a function, here we are just returning Promise.
But the caller program should use `async/await`

File: README.md [hunk 2]
```diff
@@ -13,638 +12,870 @@ npm install archethic
...
-  .send(tx, 'http://testnet.archethic.net', 60)
-  ```
+const archethic = new Archethic("https://testnet.archethic.net");
+await archethic.connect();
+
+const keychain = await archethic.account.getKeychain(accessKeychainSeed);
+const genesisUCOAddress = keychain.deriveAddress("uco", 0);
+```
+
+#### deriveKeypair(service, index)
```
[the remark this answers is not in the document]
Comment (samuelmanzanera) [reply README.md:142]: I think the `keypair` is obvious in terms of meaning.

File: lib/api.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (samuelmanzanera) [reply lib/api.js:10]: But we can mention it for more precision

**Review by Neylix — APPROVED** [review]

**Pull request merged**